### ISSUE TITLE
PropertyLabel component should be an HTML <label>

### ISSUE THREAD

**Issue by jcoyne** [LD4P/sinopia_editor#2494]
This is an accessiblity concern. All fields should have visible labels.

**Comment by jermnelson** [comment]
All of the componets depend on the `PropertLabel` to display the label. Currently we are using a `<span>`, this should be changed to a `<label>` with a **for** attribute.

**Comment by jmartin-sul** [comment]
implementation for this ticket partially working, closing in favor of #2683, which captures what's not yet working.
